syntax="proto3";
package eos.console;

message FsckProto {
  message ConfigProto {
    string key = 1;
    string value = 2;
  }

  message ReportProto {
    bool display_per_fs = 1;
    bool display_fxid = 2;
    bool display_lfn = 3;
    bool display_json = 4;
    repeated string tags = 5;
  }

  message RepairProto {
    bool async = 1;
    uint64 fid = 2;
    uint64 fsid_err = 3;
    string error = 4;
  }

  message CleanOrphansProto {
    uint64 fsid = 1;
  }

  oneof subcmd {
    bool stat = 1;
    ConfigProto config = 2;
    ReportProto report = 3;
    RepairProto repair = 4;
    CleanOrphansProto clean_orphans = 5;
  }
}
